docs/proposal/20240308-vpc-infra-creation.md

```diff
@@ -0,0 +1,393 @@
+# Dynamically create infrastructure required for VPC cluster
+
+## Motivation
+Currently, inorder to create a VPC cluster using cluster api we need to create
+resources as prerequisites. This involves work which is limiting the true
+capabilities of cluster api.
+
+
+## Goal
+1. Dynamically create the required cloud resources as a part of cluster
+   creation.
+
+
+## Proposal
+
+
+### Cluster API VPC cluster components
+
+![vpc-cluster-components.png](../images/vpc-cluster-components.png)
+
+
+### Proposed API changes
+
+
+#### VPCCluster improvements
+```go
+// IBMVPCClusterSpec defines the desired state of IBMVPCCluster.
+type IBMVPCClusterSpec struct {
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// controlPlaneEndpoint represents the endpoint used to communicate with the control plane.
+	// +optional
+	ControlPlaneEndpoint capiv1beta1.APIEndpoint `json:"controlPlaneEndpoint"`
+
+	// controlPlaneLoadBalancer is optional configuration for customizing control plane behavior.
+	// +optional
+	ControlPlaneLoadBalancer []*VPCLoadBalancerSpec `json:"controlPlaneLoadBalancer,omitempty"`
```

Comment:
This filed is already exist in current spec, so it will be a breaking change to make it slice.

Reply:
Oh, sorry, missed that change before making the change to deprecated fields. Will attempt to redesign.

Reply:
Fixed, added new field, restored and marked this one as deprecated.